src/EFCore.PG/Query/Expressions/Internal/CollateExpression.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Linq.Expressions;
+using System.Text.RegularExpressions;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Query;
+using Microsoft.EntityFrameworkCore.Query.SqlExpressions;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Query.Internal;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.Expressions.Internal
+{
+    public class CollateExpression : SqlExpression, IEquatable<CollateExpression>
+    {
+        /// <summary>
+        /// The operand on which to apply the explicit collation.
+        /// </summary>
+        [NotNull]
+        public virtual SqlExpression Operand { get; }
+
+        /// <summary>
+        /// The collation to apply to <see cref="Operand"/>.
+        /// </summary>
+        [NotNull]
+        public virtual string Collation { get; }
+
+        public CollateExpression([NotNull] SqlExpression operand, [NotNull] string collation)
+            : base(operand.Type, operand.TypeMapping)
```

Comment:
Is this TypeMapping valid? Does the typeMapping print out literals in correction collation?

Reply:
In this PR, the type mapping doesn't know anything about the collation, since it's not really part of the type name, but of the column spec (see https://github.com/dotnet/efcore/issues/11896#issuecomment-604849819). So with this PR, the provider only ever emits COLLATE:

1. In column specs DDL (e.g. when creating a new column)
2. When EF.Functions.ApplyCollation (soon to be Collate) is applied explicitly
3. When string.Equals(s, StringComparison.OrdinalIgnoresCase) is used (or Ordinal).

Therefore the type mapping doesn't seem to be concerned. Tell me if you see an issue with this approach.